src/wakefs/squashfuse_helper.cpp

```diff
@@ -0,0 +1,67 @@
+#define _XOPEN_SOURCE 700
+#define _POSIX_C_SOURCE 200809L
+
+#include "squashfuse_helper.h"
+
+#include <errno.h>
+#include <fcntl.h>
+#include <libgen.h>
+#include <poll.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <wcl/defer.h>
+#include <wcl/result.h>
+#include <wcl/xoshiro_256.h>
+
+#include <iostream>
+
+void cleanup_fifo(std::string squashfuse_fifo_path, int squashfuse_notify_pipe_fd) {
+  close(squashfuse_notify_pipe_fd);
+  unlink(squashfuse_fifo_path.c_str());
+}
```

Comment:
should be static

edit: or better yet use wcl::unique_fd as suggested below and then you can delete this